src/components/navs/AppNav/AppNavNetworkSelect.vue

```diff
@@ -1,66 +1,106 @@
+import { useRouter } from 'vue-router';
+
+import i18n from '@/plugins/i18n';
 
 import useBreakpoints from '@/composables/useBreakpoints';
-import ConfigService from '@/services/config/config.service';
+import useNetwork from '@/composables/useNetwork';
+import useNotifications from '@/composables/useNotifications';
 
 export interface NetworkOption {
   id: string;
   name: string;
   subdomain?: string;
+  networkSlug?: string;
   key?: string;
 }
 
-// SERVICES
-const configService = new ConfigService();
-
 // COMPOSABLES
 const { upToLargeBreakpoint } = useBreakpoints();
+const { networkId, networkConfig } = useNetwork();
+const router = useRouter();
+const { addNotification } = useNotifications();
 
 const networks = ref([
   {
     id: 'ethereum',
     name: 'Ethereum',
     subdomain: 'app',
+    networkSlug: 'ethereum',
     key: '1',
   },
   {
     id: 'polygon',
     name: 'Polygon',
     subdomain: 'polygon',
+    networkSlug: 'polygon',
     key: '137',
   },
   {
     id: 'arbitrum',
     name: 'Arbitrum',
     subdomain: 'arbitrum',
+    networkSlug: 'arbitrum',
     key: '42161',
   },
 ]);
 
-const appNetworkSupported = networks.value
-  .map(network => network.key)
-  .includes(configService.network.key);
+// COMPUTED
+const appNetworkSupported = computed((): boolean => {
+  return networks.value
+    .map(network => network.key)
+    .includes(networkId.value.toString());
+});
 
-const activeNetwork = computed(() =>
+const activeNetwork = computed((): NetworkOption | undefined =>
   networks.value.find(network => {
-    if (!appNetworkSupported && network.id === 'ethereum') return true;
+    if (!appNetworkSupported.value && network.id === 'ethereum') return true;
     return isActive(network);
   })
 );
 
+// LIFECYCLE
+onMounted(async () => {
+  await router.isReady();
+  if (router.currentRoute.value.query?.poolNetworkAlert) {
+    addNotification({
+      type: 'error',
+      title: '',
+      message: `${i18n.global.t('poolDoesntExist')} ${
+        router.currentRoute.value.query.poolNetworkAlert
+      }`,
+    });
+    router.replace({ query: {} });
+  }
+});
+
 // METHODS
 function iconSrc(network: NetworkOption): string {
   return require(`@/assets/images/icons/networks/${network.id}.svg`);
 }
 
 function appUrl(network: NetworkOption): string {
-  return `https://${network.subdomain}.balancer.fi`;
+  if (
+    router.currentRoute.value.name === 'pool' ||
+    router.currentRoute.value.name === 'create-pool' ||
+    router.currentRoute.value.name === 'invest' ||
+    router.currentRoute.value.name === 'withdraw' ||
+    router.currentRoute.value.name === 'migrate-pool'
+  )
+    return `/#/${network.networkSlug}?poolNetworkAlert=${networkConfig.name}`;
+
+  const currentRoute = router.currentRoute.value;
+  return router.resolve({
+    name: currentRoute.name ?? 'home',
+    params: { ...currentRoute.params, networkSlug: network.networkSlug },
+    query: currentRoute.query,
+  }).href;
 }
```

Comment:
I imagine this function is actually no longer relevant? It was only used before to get the app domain for a given network, not the path.

Reply:
It performs a different role now, for pool routes it redirects to "home" and for all other routes it updates the current url with new network.

Reply:
Let's rename it then. Maybe something like `transformUrl` or `handleNetworkChange`?